kubernetes/utils/delete_from_yaml.py

```diff
@@ -0,0 +1,146 @@
+                except FailToDeleteError as failure:
+                    # if error is returned add to failures list
+                    failures.extend(failure.api_exceptions)
+            if failures:
+                #display the error 
+                raise FailToDeleteError(failures)
+
+def delete_from_dict(k8s_client,yml_document, verbose,namespace="default",**kwargs):
+    """
+    Perform an action from a dictionary containing valid kubernetes
+    API object (i.e. List, Service, etc).
+    Input:
+    k8s_client: an ApiClient object, initialized with the client args.
+    data: a dictionary holding valid kubernetes objects
+    verbose: If True, print confirmation from the create action.
+        Default is False.
+    namespace: string. Contains the namespace to create all
+        resources inside. The namespace must preexist otherwise
+        the resource creation will fail. If the API object in
+        the yaml file already contains a namespace definition
+        this parameter has no effect.
+    Raises:
+        FailToDeleteError which holds list of `client.rest.ApiException`
+        instances for each object that failed to delete.
+    """
+    api_exceptions = []
+    try:
+        # call function delete_from_yaml_single_item 
+        delete_from_yaml_single_item(
+            k8s_client, yml_document, verbose, namespace=namespace, **kwargs
+        )
+    except client.rest.ApiException as api_exception:
+        api_exceptions.append(api_exception)
+
+    if api_exceptions:
+        raise FailToDeleteError(api_exceptions)
+
+
+def delete_from_yaml_single_item(k8s_client, yml_document, verbose=False, **kwargs):
+    # get group and version from apiVersion
+    group,_,version = yml_document["apiVersion"].partition("/")
+    if version == "":
+        version = group
+        group = "core"
+    # Take care for the case e.g. api_type is "apiextensions.k8s.io"
+    group = "".join(group.rsplit(".k8s.io", 1))
+    # convert group name from DNS subdomain format to
+    # python class name convention
+    group = "".join(word.capitalize() for word in group.split('.'))
+    func = "{0}{1}Api".format(group, version.capitalize())
+    k8s_api = getattr(client, func)(k8s_client)
+    kind = yml_document["kind"]
+    kind = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', kind)
+    kind = re.sub('([a-z0-9])([A-Z])', r'\1_\2', kind).lower()
+
+    if getattr(k8s_api,"delete_namespaced_{}".format(kind)):
+        # load namespace if provided in yml file
+        if "namespace" in yml_document["metadata"]:
+            namespace = yml_document["metadata"]["namespace"]
+            kwargs["namespace"] = namespace
+        # take name input of kubernetes object
+        name = yml_document["metadata"]["name"]
+        #call function to delete from namespace
+        res = getattr(k8s_api,"delete_namespaced_{}".format(kind))(
+         name=name,body=client.V1DeleteOptions(propagation_policy="Foreground", grace_period_seconds=5),**kwargs)
```

Comment:
if we want to achieve parity with `kubectl delete -f`, we should probably use Background policy by default with a Cascade option: https://github.com/kubernetes/kubernetes/blob/0c642b6ef01071219b9b9e34ad6d89cb9b4b1971/staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go#L304-L308

Reply:
have updated this to use background policy